mock.go

```diff
@@ -38,12 +35,24 @@ func (m *Mock) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var mr *mockResponse
 	m.Lock()
 	defer m.Unlock()
+
+	var matches []*mockResponse
 	for _, v := range m.mockResponses {
-		if v.path == path && v.method == method && v.checkFilter(r) {
+		if v.path == path && v.method == method && !v.isDepleted() {
+			matches = append(matches, v)
+			continue
+		}
+	}
+
+	matches = m.withFiltersFirst(matches)
+
+	for _, v := range matches {
+		if v.checkFilter(r) {
 			mr = v
 			break
 		}
 	}
+
 	if mr == nil {
 		w.WriteHeader(http.StatusNotFound)
```

Comment:
if isDepleted we will just return 404. Do we want some kind of assert/error if its depleted so the user knows it should use Times(X)... ?

Reply:
If it is depleted it will continue to look for the next matching mock until no more matching mocks are setup